gs_quant/api/gs/data.py

```diff
@@ -385,3 +391,48 @@ def coordinates_data_series(
             return ret[0]
         else:
             return ret
+
+    @staticmethod
+    @cachetools.cached(TTLCache(ttl=3600, maxsize=128))
+    def get_types(dataset_id: str):
+        results = GsSession.current._get(f'/data/catalog/{dataset_id}')
+        if results.get("fields"):
+            field_types = {}
+            for key, value in results.get("fields").items():
+                field_type = value.get('type')
```

Comment:
results.get("fields") is called twice. More efficient (and pythonic) would be:

if 'fields' in results:
...
    for key, value in results['fields'].items()

Alternatively, call results.get() once and assign to a variable